nima/http2/webserver/src/test/java/io/helidon/nima/http2/webserver/ConnectionConfigTest.java

```diff
@@ -103,6 +84,65 @@ void testProviderConfigBuilder() {
         assertThat(conn.serverSettings().value(Http2Setting.MAX_HEADER_LIST_SIZE), is(2048L));
     }
 
+    // Verify that HTTP/2 MAX_CONCURRENT_STREAMS is properly configured from builder
+    @Test
+    void testConfigMaxConcurrentStreams()
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+
+        // This will pick up application.yaml from the classpath as default configuration file
+        Config config = Config.create();
+        List<ServerConnectionSelector> providers = getProviders(WebServer.builder().config(config.get("server")));
+
+        // Check whether at least one Http2ConnectionProvider was found
+        Http2ConnectionSelector selector = null;
+
+        for (ServerConnectionSelector provider : providers) {
+            if (provider instanceof Http2ConnectionSelector) {
+                selector = (Http2ConnectionSelector) provider;
+            }
+        }
+        assertThat(selector, is(not(nullValue())));
+        Http2Connection conn = (Http2Connection) selector.connection(mockContext());
+        // Verify value to be updated from configuration file
+        assertThat(conn.config().maxConcurrentStreams(), is(16384L));
+        // Verify Http2Settings value to be updated from configuration file
+        assertThat(conn.serverSettings().value(Http2Setting.MAX_CONCURRENT_STREAMS), is(16384L));
+    }
+
+    // Verify that HTTP/2 validatePath is properly configured from builder
+    @Test
+    void testConfigValidatePath()
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+
+        // This will pick up application.yaml from the classpath as default configuration file
+        Config config = Config.create();
+        List<ServerConnectionSelector> providers = getProviders(WebServer.builder().config(config.get("server")));
+
+        // Check whether at least one Http2ConnectionProvider was found
+        Http2ConnectionSelector selector = null;
+
+        for (ServerConnectionSelector provider : providers) {
+            if (provider instanceof Http2ConnectionSelector) {
+                selector = (Http2ConnectionSelector) provider;
+                break;
+            }
+        }
+        assertThat(selector, is(not(nullValue())));
+        Http2Connection conn = (Http2Connection) selector.connection(mockContext());
+        // Verify value to be updated from configuration file
+        assertThat(conn.config().validatePath(), is(false));
+    }
+
+    // Retrieve ServerConnectionSelector instances from WebServer.Builder
+    @SuppressWarnings("unchecked")
+    private List<ServerConnectionSelector> getProviders(WebServer.Builder wsBuilder)
```

Comment:
Please do not do this. Using reflection to get a private method to retrieve information is not a good practice. This will escape refactoring and will cause trouble in the future.

Reply:
Tests rewritten to avoid using reflection.